libbeat/metric/system/cgroup/cgv2/io_helper_linux.go

```diff
@@ -48,13 +51,14 @@ func fetchDeviceName(major, minor uint64) (bool, string, error) {
 		if !ok {
 			return nil
 		}
-		devID := infoT.Rdev
+		devID = uint64(infoT.Rdev)
+
 		// do some bitmapping to extract the major and minor device values
 		// The odd duplicated logic here is to deal with 32 and 64 bit values.
 		// see bits/sysmacros.h
-		curMajor := ((devID & 0xfffff00000000000) >> 32) | ((devID & 0x00000000000fff00) >> 8)
-		curMinor := ((devID & 0x00000000000000ff) >> 0) | ((devID & 0x00000ffffff00000) >> 12)
-		if curMajor == major && curMinor == minor {
+		curMajor = ((devID & 0xfffff00000000000) >> 32) | ((devID & 0x00000000000fff00) >> 8)
+		curMinor = ((devID & 0x00000000000000ff) >> 0) | ((devID & 0x00000ffffff00000) >> 12)
+		if uint64(curMajor) == uint64(major) && uint64(curMinor) == uint64(minor) {
```

Comment:
I'm not sure these explicit casts to `uint64()` are necessary here?

Reply:
I think your correct.